fplll/hnf.cpp

```diff
@@ -0,0 +1,346 @@
+/* Copyright (C) 2018 Arnaud Sipasseuth. (inspiration from FLINT (2017 version))
```

Comment:
The default API of fplll is "make an HNF object and then call that". I appreciate this isn't quite needed here but it makes sense to follow the API convention anyway (?)

Reply:
I tried to look similar to others but I will remove that.